pkg/prompts/prompts.go

```diff
@@ -153,9 +171,19 @@ func RunDefaultableStringPrompt(customPrompt config.BuilderVar, defaultValue str
 	if err != nil {
 		return "", err
 	}
-	// Variable-level substitution, we need to get defaults so later references can be resolved in this loop
-	if input == "" && defaultString != "" {
-		input = defaultValue
+	if customPrompt.Name == "APPNAME" {
+		// Sanitize the user input
+		sanitizedInput := sanitizeAppName(input)
+		if sanitizedInput == "" {
+			// If sanitized input is empty, use the sanitized directory-based name
+			sanitizedInput = defaultValue
+		}
+		input = sanitizedInput
+	} else {
+		// Variable-level substitution, we need to get defaults so later references can be resolved in this loop
+		if input == "" && defaultString != "" {
```

Comment:
why doesn't this logic apply to app name?

Reply:
this is checked in the if statement and since I'm sanitizing the appname I'm handling it separately.